pkg/build/util/generate.go

```diff
@@ -154,3 +160,19 @@ func SubstituteImageReferences(build *buildapi.Build, oldImage string, newImage
 		}
 	}
 }
+
+// GetBuildPodName returns name of the build pod.
+func GetBuildPodName(build *buildapi.Build) string {
+	return fmt.Sprintf("pod-%s", build.Name)
+}
+
+// GetNextBuildName returns name of the next build and increments BuildConfig's LastVersion.
+func GetNextBuildName(bc *buildapi.BuildConfig) string {
+	bc.LastVersion++
```

Comment:
this is a race condition.  if i create multiple builds at the same time, they could all end up with the same "lastversion" value since they all start from the same BuildConfig (assuming they all did a get on buildconfig before any one of them got to the point of incrementing this value and calling Update on the BC).

you need some sort of universal locking based on the BC here.  (and even that has clustering/scaling limitations but i think we're ok there)